File: 2021/05/18/main.py
class Solution:
    def findDuplicate(self, paths: List[str]) -> List[List[str]]:
        d = {}
        for p in paths:
            ps = p.split(" ")
            for f in ps[1:]:
                fs = f.split("(")
                n, c = fs[0], fs[1][:-1]
                if not c in d:
                    d[c] = []
                d[c].append(ps[0] + "/" + n)
        return [p for p in d.values() if len(p) > 1]

File: 2021/05/18/main.go
package main

import (
	"strings"
)

func findDuplicate(paths []string) [][]string {
	d := make(map[string][]string, len(paths)*2)
	for _, p := range paths {
		ps := strings.Split(p, " ")
		for _, f := range ps[1:] {
			fs := strings.Split(f, "(")
			n, c := fs[0], fs[1][:len(fs[1])-1]
			d[c] = append(d[c], ps[0]+"/"+n)
		}
	}
	res := make([][]string, 0, len(d))
	for _, p := range d {
		if len(p) > 1 {
			res = append(res, p)
		}
	}
	return res
}
